src/components/Modal/CartProduct/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useContext } from 'react';
+import ProductsContext from '../../../context/products-context';
+import CartContext from '../../../context/cart-context';
+import ErrorContext from '../../../context/error-context';
+import isStorageSupported from '../../../helpers/isStorageSupported';
+
+import { SingleWrap, CloseBtn, ImgWrap, ProductImg, DataWrap, UpperWrap, TitleWrap, DownerWrap, PriceWrap, TotalWrap } from './style';
+import { ProductObj, CartItem } from '../../../types/interfaces';
+
+interface Props {
+	cartItem: CartItem;
+	onClickClose: any;
+}
+
+const CartProduct = ({cartItem, onClickClose}: Props) => {
+	const productsCtx = useContext(ProductsContext);
+	const cartCtx = useContext(CartContext);
+	const errorCtx = useContext(ErrorContext);
+	const [quantity, setQuantity] = useState<number>(cartItem.quantity);
+
+	const currentProduct = productsCtx.products.filter((singleProduct: ProductObj) => singleProduct.id === cartItem.id);
+
+	const onChangeHandler = (e: React.FormEvent<HTMLInputElement>) => {
+		const id = parseInt(e.currentTarget.id);
+		const newValue = parseInt(e.currentTarget.value);
+		const price = parseFloat(e.currentTarget.name);
+		const totalPrice = price * newValue;
+		setQuantity(newValue);
+		const newCartItems = cartCtx.items.map((item: CartItem) => item.id === id ? { ...item, quantity: newValue, price: price, total: totalPrice} : item);
```

Comment:
although this works, in React, it's recommended to pass the data directly from the object or state when it is already available, rather than extracting it from the eventHandler.